src/main/java/org/mule/service/http/impl/service/server/grizzly/ResponseStreamingCompletionHandler.java

```diff
@@ -103,7 +104,10 @@ private int calculateBufferSize(FilterChainContext ctx, ClassLoader ctxClassLoad
   }
 
   public void start() throws IOException {
-    sendInputStreamChunk();
+    withContextClassLoader(getCtxClassLoader(), () -> {
```

Comment:
avoid using these functional idioms. Those are deprecated and introduce performance overhead

Reply:
you beat me by two minutes.... This should be done using the `org.mule.runtime.core.api.util.ClassUtils#setContextClassLoader` method. Look at the commit that introduce that for examples of how to properly use it.

Also, keep in mind that even with this method this will introduce a performance regression. Is the bug worth it?

Reply:
Ok, I changed the deprecated functional idioms for ClassUtils#setContextClassLoader as @marianogonzalez suggested.
Regarding the fix itself, if you think that using a different CL doesn't matter in this case, then we can forget this fix.  Perhaps we can measure the performance in both scenarios to decide if the change worth it?

Reply:
Checking the performance impact would be ideal. Another thing you can do is to only perform the context switch if the `org.mule.runtime.api.util.MuleSystemProperties#MULE_LOG_SEPARATION_DISABLED` system property is not present. That would make this transparent, performance job wise

Reply:
I've made the changes being aware of MULE_LOG_SEPARATION_DISABLED and also run two performance tests using the same build with and without the patch.  Here you can see the results: https://docs.google.com/spreadsheets/d/1IWu3jzTE5zaqYeJmXOCYwROm2iNGJ787YJMv1bpyzDA/edit?usp=sharing
It seems that TPS lowered fewer than 1%, and the average response time has also increased less than 1%.  What do you think?